components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/utils/SCIMCommonUtils.java

```diff
@@ -304,4 +304,54 @@ public static boolean isFilteringEnhancementsEnabled() {
         return Boolean.parseBoolean(IdentityUtil.getProperty
                 (SCIMCommonConstants.SCIM_ENABLE_FILTERING_ENHANCEMENTS));
     }
+
+    /**
+     * Checks whether the identity.xml config is available for applying filtering enhancements. This makes sure that Eq
+     * should strictly check for the String match (in this case cross user store search should not be performed).
+     *
+     * @return whether 'FilterUsersAndGroupsOnlyFromPrimaryDomain' property is enabled in identity.xml.
+     */
+    public static boolean isFilterUsersAndGroupsOnlyFromPrimaryDomainEnabled() {
+
+        return Boolean.parseBoolean(IdentityUtil
+                .getProperty(SCIMCommonConstants.SCIM_ENABLE_FILTER_USERS_AND_GROUPS_ONLY_FROM_PRIMARY_DOMAIN));
+    }
+
+    /**
+     * Checks whether the identity.xml config is available for prepending the 'PRIMARY/' in each role and
+     * which belong to Primary domain in the response of Groups endpoints.
+     *
+     * @return whether 'MandateDomainForGroupNamesInGroupsResponse' property is enabled in identity.xml.
+     */
+    public static boolean mandateDomainForGroupNamesInGroupsResponse() {
+
+        return Boolean.parseBoolean(IdentityUtil
+                .getProperty(SCIMCommonConstants.SCIM_ENABLE_MANDATE_DOMAIN_FOR_GROUPNAMES_IN_GROUPS_RESPONSE));
+    }
+
+    /**
+     * Checks whether the identity.xml config is available for prepending the 'PRIMARY/' in each role and
+     * username which belong to Primary domain in the responses of both Users and Groups endpoints.
+     *
+     * @return whether 'MandateDomainForUsernamesAndGroupNamesInResponse' property is enabled in identity.xml.
+     */
+    public static boolean mandateDomainForUsernamesAndGroupNamesInResponse() {
+
+        return Boolean.parseBoolean(IdentityUtil
+                .getProperty(SCIMCommonConstants.SCIM_ENABLE_MANDATE_DOMAIN_FOR_USERNAMES_AND_GROUPNAMES_IN_RESPONSE));
+    }
+
+    /**
+     * Method to prepend PRIMARY domain name to given String when that String not contains DOMAIN_SEPARATOR.
+     * @param value
+     * @return Given String if given String not contains DOMAIN_SEPARATOR, String with PRIMARY domain otherwise.
+     */
+    public static String prependPrimaryDomain(String value) {
```

Comment:
Shall we change the method name to reflect the internals? This name does not say that the `PRIMARY` would not be appended if already a domain name is present

Reply:
Fixed